agent/src/main/java/io/pyroscope/javaagent/JFRProfilerDelegate.java

```diff
@@ -0,0 +1,126 @@
+package io.pyroscope.javaagent;
+
+import io.pyroscope.http.Format;
+import io.pyroscope.javaagent.config.Config;
+import io.pyroscope.labels.Pyroscope;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class JFRProfilerDelegate implements ProfilerDelegate {
+    private static final String RECORDING_NAME = "pyroscope";
+    private Config config;
+    private File tempJFRFile;
+    private Path jcmdBin;
+
+    public JFRProfilerDelegate(Config config) {
+        setConfig(config);
+    }
+
+    public void setConfig(final Config config) {
+        this.config = config;
+        jcmdBin = findJcmdBin();
+        try {
+            // flight recorder is built on top of a file descriptor, so we need a file.
+            tempJFRFile = File.createTempFile("pyroscope", ".jfr");
+            tempJFRFile.deleteOnExit();
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    /**
+     * Start JFR profiler
+     */
+    public synchronized void start() {
+        try {
+            List<String> commands = new ArrayList<>();
+            commands.add(jcmdBin.toString());
+            commands.add(String.valueOf(CurrentPidProvider.getCurrentProcessId()));
+            commands.add("JFR.start");
+            commands.add("name=" + RECORDING_NAME);
+            commands.add("filename=" + tempJFRFile.getAbsolutePath());
+            commands.add("settings=pyroscope");
+            ProcessBuilder processBuilder = new ProcessBuilder(commands);
+            Process process = processBuilder.start();
```

Comment:
I get 
```
6108:
Could not parse settings file 'pyroscope'
```

do we need to unpack this file?

Can we somehow detect if profiling failed to start?

Reply:
oh, forgot, to add description to README, this requires profiler settings to be copied to lib/jfr dir in JDK

Reply:
I dont like this manual step, this should be automatic. It does seem to work if I specify absolute path to the settings file, see my comment bellow. Can we unpack this settings somewhere safely and provide absolute path to it, so that users dont have to do anything? Would that work?

Reply:
In new commit I have added extraction of settings file from jar to temp location and use it as a absolute path in jcmd options, works like a charm, and doesn't require any extra steps fro users :)